app/Models/Internet/MacAddress.php

```diff
@@ -33,15 +55,49 @@ class MacAddress extends Model
         'user_id', 'mac_address', 'comment', 'state', 'ip',
     ];
 
+    // Default attributes
     protected $attributes = [
-        'comment' => '',
         'state' => self::REQUESTED,
     ];
 
+    // Return translated state every time the model is retrieved.
+    protected $appends = ['translated_state'];
 
-    public function user(): BelongsTo
+    /**
+     * Get the translated_state attribute.
+     *
+     * @return Attribute
+     */
+    public function translatedState(): Attribute
+    {
+        return Attribute::make(
+            get: fn () => __('internet.' . strtolower($this->state))
+        );
+    }
+
+    /**
+     * The user's internet access model.
+     *
+     * @return BelongsTo
+     */
+    public function internetAccess(): BelongsTo
+    {
+        return $this->belongsTo(InternetAccess::class, 'user_id', 'user_id');
+    }
+
+    /**
+     * The user that this mac address belongs to.
+     */
+    public function user(): \Illuminate\Database\Eloquent\Relations\HasOneThrough
     {
-        return $this->belongsTo(User::class);
+        return $this->hasOneThrough(
+            User::class,
+            InternetAccess::class,
+            'user_id', // Foreign key on InternetAccess table...
+            'id', // Foreign key on Users table...
+            'user_id', // Local key on MacAddresses table...
+            'user_id' // Local key on InternetAccess table...
+        );
```

Comment:
Why? Do you want to treat the `MacAddress` as if it belonged to the `InternetAccess` even though it is not reflected in the database?

Reply:
Yes. I know its a bit strange here but I don't think this would cause problems

Reply:
It's a bit strange and probably heavier on the queries, but I don't think it is a problem. It mainly was just a question.